### PR TITLE
Add new target for uploading binaries to release

### PR DESCRIPTION
**Pull request by vadim-gleyf**
### what
* Add new `github/push-artifacts` target

### why
* For uploading artifacts in a codefresh build step

### CODE REVIEW

File: bin/github_push_artifacts.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+# PREREQUISITES
+#
+# ghr
+
+TAG=$1
+ARTIFACTS=$2
+GITGUB_TOKEN=$3
+
+ghr -t $GITGUB_TOKEN $TAG $ARTIFACTS
```
Comment (osterman) [inline bin/github_push_artifacts.sh]: I don't see any benefit with having a shell script here. Use shell scripts when there is complicated bash logic/flow control.  Use inline commands in make targets when there's just a one-liner.

**Comment by osterman** [comment]
This PR should be depended on another one in packages, but I don't see any here: 

https://github.com/cloudposse/packages/pulls

**Review by osterman — CHANGES_REQUESTED** [review]
See inline comments

File: modules/github/Makefile.release [hunk 2]
```diff
@@ -19,3 +19,9 @@ github/download-public-release:
 	$(call assert-set,VERSION)
 	$(call assert-set,OUTPUT)
 	$(BUILD_HARNESS_PATH)/bin/github_download_public_release.sh $(VERSION) $(OUTPUT)
+
```
Comment (osterman) [inline modules/github/Makefile.release:22]: Add a target description. e.g.
```
## Push all release artifacts to GitHub
```

File: modules/github/Makefile.release [hunk 3]
```diff
@@ -19,3 +19,9 @@ github/download-public-release:
...
+github/push-artifacts:
```
Comment (osterman) [inline modules/github/Makefile.release]: Add a package dependency here:

```
github/push-artifacts: packages/install/ghr
```

File: modules/github/Makefile.release [hunk 4]
```diff
@@ -19,3 +19,10 @@ github/download-public-release:
 	$(call assert-set,VERSION)
 	$(call assert-set,OUTPUT)
 	$(BUILD_HARNESS_PATH)/bin/github_download_public_release.sh $(VERSION) $(OUTPUT)
+
+## Push all release artifacts to GitHub
+github/push-artifacts: packages/install/ghr
+	$(call assert-set,GIT_TAG)
+	$(call assert-set,ARTIFACTS)
+	$(call assert-set,GITHUB_TOKEN)
+	@ghr -t $(GITHUB_TOKEN) $(GIT_TAG) $(ARTIFACTS)
```
Comment (osterman) [inline modules/github/Makefile.release]: Can `ghr` work without passing the token as a parameter? E.g. can it look in the environment? Would rather not pass it on the command line if we can avoid it.

**Review by osterman — COMMENTED** [review]
See inline comments

**Review by goruha — CHANGES_REQUESTED** [review]
Update package version to use https://github.com/cloudposse/packages/pull/27

**Review by osterman — CHANGES_REQUESTED** [review]
See comment by @goruha

**Comment by vadim-gleyf** [comment]
#### Depend on 
- https://github.com/cloudposse/packages/pull/27

File: modules/github/Makefile.release [hunk 5]
```diff
@@ -19,3 +19,11 @@ github/download-public-release:
 	$(call assert-set,VERSION)
 	$(call assert-set,OUTPUT)
 	$(BUILD_HARNESS_PATH)/bin/github_download_public_release.sh $(VERSION) $(OUTPUT)
+
+## Push all release artifacts to GitHub
+## To use ghr, you need a GitHub API token.
```
Comment (osterman) [inline modules/github/Makefile.release]: Target annotations `##` must be on one-line and cannot repeat. Also, they must immediately precede the target name. This is due to how it's implemented with `awk`: https://github.com/cloudposse/build-harness/blob/master/Makefile.helpers#L39-L51

Instead, notes to developers should just use a single `#`. See other examples in makefiles.

File: modules/github/Makefile.release [hunk 6]
```diff
@@ -19,3 +19,11 @@ github/download-public-release:
 	$(call assert-set,VERSION)
 	$(call assert-set,OUTPUT)
 	$(BUILD_HARNESS_PATH)/bin/github_download_public_release.sh $(VERSION) $(OUTPUT)
+
+## Push all release artifacts to GitHub
```
Comment (osterman) [inline modules/github/Makefile.release]: Perhaps write: 
```
## Push all release artifacts to GitHub (Required: `GITHUB_TOKEN`)
```

**Review by osterman — CHANGES_REQUESTED** [review]
See inline comments

**Review by goruha — APPROVED** [review]

**Review by osterman — APPROVED** [review]

**Comment by osterman** [comment]
https://github.com/cloudposse/build-harness/issues/82

**Pull request merged**